port/unix/omrsysinfo.c

```diff
@@ -1091,13 +1093,20 @@ omrsysinfo_get_number_CPUs_by_type(struct OMRPortLibrary *portLibrary, uintptr_t
 		if (0 == toReturn) {
 			Trc_PRT_sysinfo_get_number_CPUs_by_type_failedOnline("(no errno) ", 0);
 		}
-#elif defined(LINUX) || defined(OSX)
+#elif (defined(LINUX) && !defined(OMRZTPF)) || defined(OSX)
 		/* returns number of online(_SC_NPROCESSORS_ONLN) processors, number configured(_SC_NPROCESSORS_CONF) may  be more than online */
 		toReturn = sysconf(_SC_NPROCESSORS_ONLN);
 
 		if (0 >= toReturn) {
 			Trc_PRT_sysinfo_get_number_CPUs_by_type_failedOnline("errno: ", errno);
 		}
+#elif defined(OMRZTPF)
+		/* USE PHYSICAL for ONLINE on z/TPF */
+		toReturn = omrsysinfo_get_number_CPUs_by_type(portLibrary, OMRPORT_CPU_PHYSICAL);
```

Comment:
inside the port library you should not be using the port library macros for making calls but rather call the function directly through the port library function table.

Reply:
@charliegracie just to confirm, instead of: omrsysinfo_get_number_CPUs_by_type(OMRPORT_CPU_PHYSICAL)
Call should be replaced with?
portLibrary->sysinfo_get_number_CPUs_by_type(portLibrary, OMRPORT_CPU_PHYSICAL);

There are other occurences in the file (not z/TPF) that made the omrsysinfo_get_number... calls (e.g. line 1211)

Reply:
Let me know if you wanted me to replace all those calls as well.